python/cells.cpp

```diff
@@ -577,8 +577,13 @@ void register_cells(pybind11::module& m) {
             return util::pprintf("<arbor.iclamp: frequency {} Hz>", c.frequency);});
 
     // arb::threshold_detector
+    struct spike_detector {};
+    pybind11::class_<spike_detector>(m, "spike_detector")
+        .def(pybind11::init<>([](pybind11::object) {
+            throw arb::arbor_exception{"spike_detector is deprecated, use threshold_detector instead."};
+        }));
 
-    pybind11::class_<arb::threshold_detector> detector(m, "spike_detector",
+    pybind11::class_<arb::threshold_detector> detector(m, "threshold_detector",
             "A spike detector, generates a spike when voltage crosses a threshold. Can be used as source endpoint for an arbor.connection.");
```

Comment:
A **threshold** detector ...

Reply:
It detects a threshold being crossed. A detector could be anything.